tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,252 @@
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, const std::vector<std::string>& inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions):
+   fInputFiles(inputFiles), fOutputFile(output), fTreeName(name)
+{
+      for (std::string expr: expressions) {
+        std::string errMessage = HandleExpressionConfig(expr);
+         if (!errMessage.empty())
+            throw std::runtime_error(errMessage + " in " +  expr);
+      }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return true if the input arguments create the output file correctly
+
+bool TSimpleAnalysis::Analyze()
+{
+   TChain chain(fTreeName.c_str());
+   for (const std::string& inputfile: fInputFiles)
+      chain.Add(inputfile.c_str());
+   TFile ofile(fOutputFile.c_str(), "RECREATE");
+
+   for (const auto &histo : fHists) {
+      chain.Draw((histo.second.first + ">>" + histo.first).c_str(), histo.second.second.c_str(), "goff");
+      TH1F *ptrHisto = (TH1F*)gDirectory->Get(histo.first.c_str());
+      ptrHisto->Write();
+      if (ptrHisto == nullptr)
+         return false;
+   }
+   return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return false if not a tree name
+///
+/// param[in] line line read from the input file
+
+static bool HandleTreeNameConfig(const std::string& line)
+{
+   return line.find("=") == std::string::npos;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Skip subsequent empty lines in a string and returns the number of the
+/// current line of the input file
+///
+/// param[in] numbLine number of the input file line
+
+std::string TSimpleAnalysis::SkipSubsequentEmptyLines(int& numbLine)
+{
+   std::string notEmptyLine;
+
+   while (getline(fIn,notEmptyLine) && DeleteSpaces(notEmptyLine)
+          && (notEmptyLine.empty() || notEmptyLine.find("#") == 0 ||
+              notEmptyLine.find_first_not_of(" ") == std::string::npos))
+      {numbLine++;}
+   numbLine++;
+   return notEmptyLine;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// After the eventual skip of subsequent empty lines, returns true if the
+/// line is a comment
+///
+/// param[in] line line read from the input file
+/// param[in] readingSection current section of the read file
+/// param[in] numbLine number of the input file line
+
+bool TSimpleAnalysis::HandleLines(std::string& line, int& readingSection, int& numbLine)
+{
+   if (line.empty() || line.find_first_not_of(" ") == std::string::npos) {
```

Comment:
Looks like DeleteSpaces() should have been called before (but that will be fixed by grouping getline and DeleteSpaces into a new function)